src/common/details/console/gui/controls/InOutPipeServer.cs

```diff
@@ -0,0 +1,63 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
+//
+
+using System.Text;
+using System.Text.Json;
+
+namespace Azure.AI.Details.Common.CLI.ConsoleGui
+{
+    public class InOutPipeServer
+    {
+        public static bool IsInOutPipeServer => _inOutPipeServer != null;
+
+        public static string? GetInputFromUser(string prompt, string? value = null)
+        {
+            using var stream = new MemoryStream();
+            using var writer = new Utf8JsonWriter(stream, new JsonWriterOptions{ Indented = false });
+            writer.WriteStartObject();
+
+            writer.WriteString("prompt", prompt);
+            if (value != null)
+            {
+                writer.WriteString("value", value);
+            }
+
+            writer.WriteEndObject();
+            writer.Flush();
+
+            var json = Encoding.UTF8.GetString(stream.ToArray());
+            Console.WriteLine($"\n{_inOutPipeServer} GetInputFromUser {json}");
+
+            return Console.ReadLine();
```

Comment:
Not a big deal now, but we should probably think about what happens if we expose a rich multi-line text box to the user for a chat scenario. Just a thought